primitives/xcm/src/transactor_traits.rs

```diff
@@ -23,18 +24,40 @@ pub enum UtilityAvailableCalls {
 	AsDerivative(u16, Vec<u8>),
 }
 
-// Trait that the ensures we can encode a call with utility functions.
-// With this trait we ensure that the user cannot control entirely the call
-// to be performed in the destination chain. It only can control the call inside
-// the as_derivative extrinsic, and thus, this call can only be dispatched from the
-// derivative account
+// The hrmp calls that need to be implemented as part of
+// this pallet
+#[derive(Debug, PartialEq, Eq)]
+pub enum HrmpAvailableCalls {
+	InitOpenChannel(ParaId, u32, u32),
+	AcceptOpenChannel(ParaId),
+	CloseChannel(HrmpChannelId),
+}
+
+// Trait that the ensures we can encode a call with utility and hrmp functions.
 pub trait UtilityEncodeCall {
+	// With this trait we ensure that the user cannot control entirely the call
+	// to be performed in the destination chain. It only can control the call inside
+	// the as_derivative extrinsic, and thus, this call can only be dispatched from the
+	// derivative account
 	fn encode_call(self, call: UtilityAvailableCalls) -> Vec<u8>;
 }
 
+// Trait that the ensures we can encode a call with hrmp functions.
+// With this trait we ensure that the user cannot control entirely the call.
+// to be performed in the destination chain.
+pub trait HrmpEncodeCall {
+	fn hrmp_encode_call(call: HrmpAvailableCalls) -> Result<Vec<u8>, XcmError>;
+}
+
+impl HrmpEncodeCall for () {
+	fn hrmp_encode_call(_call: HrmpAvailableCalls) -> Result<Vec<u8>, XcmError> {
+		Err(XcmError::Unimplemented)
+	}
+}
+
 // Trait to ensure we can retrieve the destination if a given type
-// It must implement UtilityEncodeCall
-// We separate this in two traits to be able to implement UtilityEncodeCall separately
+// It must implement RelayEncodeCall
```

Comment:
This is not consistent with the trait name `UtilityEncodeCall`, or maybe you missed to create the marker trait `RelayEncodeCall`?

Reply:
Nah it's just I changed the trait name a million times haha, this is just a leftover